### PR TITLE
Fix truncated descriptions in docs

### LINKED ISSUE #1550: Ability to overwrite existing property docs with CloudFormation docs
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

We currently augment missing docs with CloudFormation docs, but we might want to create the ability to selectively pick the CloudFormation docs if those are better.

### PR DESCRIPTION
**Pull request by flostadler**
The documentation of some resources/properties is incomplete or incorrect in the AWS CloudFormation schema.
The CloudFormation docs are more complete and accurate in these cases. This change adds the ability to choose
for which resources the CloudFormation docs should be preferred.

It enrolls the following resources/properties in the CloudFormation docs:
- AWS::EC2::Volume
- AWS::EC2::VPCEndpoint
- AWS::ECR::Repository.EncryptionConfiguration.EncryptionType
- AWS::IAM::Role.Policies
- AWS::IAM::Role.RoleName

We cannot automatically match which descriptions should get replaced by the CloudFormation docs because the two have subtle differences (extra/different punctuation marks, CFN vs AWS CloudFormation, etc.)

Fixes #1572
Fixes #1550

### CODE REVIEW

File: provider/pkg/schema/gen.go [hunk 1]
```diff
@@ -983,6 +994,11 @@ func (ctx *cfSchemaContext) updateDesc(refName, propName string, spec *jsschema.
 				}
 			}
 		}
+	} else if _, ok := forceDocumentationAugmentation[GetFullyQualifiedPropertyName(refName, propName)]; ok {
+		if desc, found := ctx.docs.GetPropertyDesc(refName, propName); found {
```
Comment (thomas11) [inline provider/pkg/schema/gen.go:998]: Else: warn or report if augmentation was configured but not found?

**Review by thomas11 — COMMENTED** [review]
Nice! A test would be good, though.

File: provider/pkg/schema/gen.go (hunk 1 above)
Comment (flostadler) [reply provider/pkg/schema/gen.go:999]: Yeah that's a good idea! I'm adding it to the report

**Review by thomas11 — APPROVED** [review]

**Comment by t0yv0** [comment]
🚢

**Pull request merged**